tofhir-server/src/main/scala/io/tofhir/server/endpoint/CodeSystemEndpoint.scala

```diff
@@ -61,7 +61,9 @@ class CodeSystemEndpoint(toFhirEngineConfig: ToFhirEngineConfig) extends LazyLog
       complete {
         service.getCodeSystem(terminologyId, codeSystemId) map {
           case Some(codeSystem) => StatusCodes.OK -> codeSystem
-          case None => StatusCodes.NotFound -> s"Concept map  with id $codeSystemId not found"
+          case None => StatusCodes.NotFound -> {
+            throw ResourceNotFound("Code system not found", s"Code system with id $codeSystemId not found")
```

Comment:
Instead of **endpoint** class, the underlying **repository** class should throw this exception. Moreover, let's add a test case for this error.

This comment also applies to the changes in following files:

- tofhir-server/src/main/scala/io/tofhir/server/endpoint/ConceptMapEndpoint.scala 
- tofhir-server/src/main/scala/io/tofhir/server/endpoint/JobEndpoint.scala
- tofhir-server/src/main/scala/io/tofhir/server/endpoint/MappingEndpoint.scala
- tofhir-server/src/main/scala/io/tofhir/server/endpoint/SchemaDefinitionEndpoint.scala
- tofhir-server/src/main/scala/io/tofhir/server/endpoint/TerminologyServiceManagerEndpoint.scala
- tofhir-server/src/main/scala/io/tofhir/server/endpoint/ProjectEndpoint.scala